### ISSUE TITLE
Panel mode z-index

### ISSUE THREAD

**Issue by somepaulo** [micheleg/dash-to-dock#563]
When switching on 'Panel mode' the dash overlaps the top panel's rounded corner.
In web development terms it's a z-index issue, what goes on top of what.
Far from being a deal breaker, it does look a bit odd though.
![d2d](https://user-images.githubusercontent.com/15643750/29297218-511787bc-8157-11e7-811b-986e2da4c3e7.png)
![d2d-2](https://user-images.githubusercontent.com/15643750/29297219-5410e0d0-8157-11e7-9019-133abf27ff9e.png)
![d2d-3](https://user-images.githubusercontent.com/15643750/29297220-570a8e3a-8157-11e7-8d65-3c9c023f8792.png)

**Comment by franglais125** [comment]
The extension purposefully hides the corners in some situations. You can see that the corner on the other side is absent as well, given some conditions.

https://github.com/micheleg/dash-to-dock/blob/master/docking.js#L1888

In the second screenshot it looks like you have enabled autohide? If so, the corners are not hidden.

**Comment by somepaulo** [comment]
Yes, that's correct, this relates to autohide mode. The corners are not hidden, but the extended dock covers the corner.

**Comment by franglais125** [comment]
Your proposal would be to have the panel have a higher priority over the dock, so that the corners are visible even with an opaque dock?

**Comment by somepaulo** [comment]
I guess so.
I think it only applies to Gnome Shell themes that use a panel with corners, like the default theme. But I do think that from a purely visual perspective, when the dock is in panel mode with autohide it would look better if the top panel had z-index priority over it.

**Comment by franglais125** [comment]
I haven't worked so much on z-priority for this extension, so off the top of my head I don't know what other problems could arise. I know there are already some elements that we have to pay attention to (legacy tray bar, windows, something else..?), so I don't know if we can push the dock to a lower z, or the panel to a higher one.

I'll simply let @micheleg weigh in.

**Comment by somepaulo** [comment]
Those few pixels are definitely not a top priority issue. More akin to a feature request if you guys ever get the time for it.
Maybe drop this off to the Ubuntu team and then pick up whatever solution they come up with in their fork.

Anyway, thanks for the great extension. Have been using it for ages. Window previews are a great addition. Would be great for them to appear on icon hover.

All the best!

**Comment by micheleg** [comment]
At the moment we keep the dock below the modal layer and untill now below the legacyTray: https://github.com/micheleg/dash-to-dock/blob/44ed947e87692935fbfe934acc0e9faca4ec5af3/docking.js#L1106:L1112

and temporary raise it further during drag-and-drop:
https://github.com/micheleg/dash-to-dock/blob/44ed947e87692935fbfe934acc0e9faca4ec5af3/docking.js#L1163:L1166

I don't think there's any thechincal reason why it could't be put just below the panel with something like this:
```Main.layoutManager.uiGroup.set_child_below_sibling(this.actor, Main.layoutManager.panelBox);```

However, visually it's stil not perfect either. For example, with the new transparent panel (3.25+), the effective color of the panel and corner do not mach as the transparency reveals either the wallpaper or the dock background, and even making the dock transparent too (#513) would results in different colors being rendered.
